pkg/scheduler/core/spreadconstraint/select_groups.go

```diff
@@ -0,0 +1,196 @@
+package spreadconstraint
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+// GroupInfo indicate the group info.
+type GroupInfo struct {
+	name   string
+	value  int
+	weight int64
+}
+
```

Comment:
People may be confused by the terms 'value' and 'weight'. Perhaps using 'clustersNum' for value and 'groupScore' for weight would be clearer.

Reply:
Yeah, they should have identical meanings. `value` is not only about the number of clusters but `regions`, `zone`, etc.
eg: some zones in a region; some region in a provider...
And 'weight' is just for 'weight', not 'score', `select_groups` should be completely out of the concept of `region`, `zone` and `provider` like a tool.